### PR TITLE
nixos/wordpress: Drop old deprecated interface

### PR DESCRIPTION
**Pull request by onny**
###### Motivation for this change
This PR drops the old deprecated interface from the Wordpress module. The former syntax
`services.wordpress."hostname.local" = {}` changed to `services.wordpress.sites."hostname.local" = {}` with NixOS 21.11.
It would be good to drop the old deprecated interface with 22.05 to cleanup the module.
The deprecation warning was introduced with 21.11 and also noted in the release notes.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/tests/wordpress.nix [hunk 1]
```diff
@@ -15,15 +15,12 @@ import ./make-test-python.nix ({ pkgs, ... }:
       services.httpd.adminAddr = "webmaster@site.local";
       services.httpd.logPerVirtualHost = true;
 
-      services.wordpress = {
-        # Test support for old interface
+      services.wordpress.sites = {
         "site1.local" = {
-          database.tablePrefix = "site1_";
+          database.tablePrefix = "site2_";
```
Comment (mohe2015) [inline nixos/tests/wordpress.nix:20]: any reason this is changed?

Comment (eonpatapon) [reply nixos/tests/wordpress.nix:20]: I think we should keep both sites in the test, only adjust to use the final interface

Comment (onny) [reply nixos/tests/wordpress.nix]: Good catch @mohe2015, fixed it!

**Review by eonpatapon — APPROVED** [review]

**Comment by onny** [comment]
Added some release notes about the removal of the old deprecated interface

File: nixos/modules/services/web-apps/wordpress.nix [hunk 2]
```diff
@@ -268,10 +262,7 @@ in
   options = {
     services.wordpress = mkOption {
       type = types.submodule {
```
Comment (eonpatapon) [inline nixos/modules/services/web-apps/wordpress.nix:264]: I think the submodule is not needed anymore, I had to use it for `freeformType`, but it's not needed anymore.

We just need:

 ```nix
services.wordpress = {
  sites = mkOption { ... };
  webserver = mkOption { ... };
};

**Comment by onny** [comment]
Thank you @eonpatapon that was a good hint, fixed it :)

**Review by eonpatapon — APPROVED** [review]

**Comment by happysalada** [comment]
Thank you!

**Pull request merged**